### ISSUE TITLE
Installation instructions don't work for me; variety of issues

### ISSUE THREAD

**Issue by GKFX** [camdram/camdram#404]
README.md currently asks for the Debian packages
```
git-core php5 php5-cli curl php5-curl php5-intl php5-sqlite php-apc php5-gd php5-json
```
On Debian Buster, `curl` depends on `libcurl4` and `php5-curl` depends on `libcurl3`. Only one of these can be installed at once. Installing without `curl` doesn't appear to cause any issues; should it be dropped from requirements?

-----
Proceeding past that to `php app/console camdram:database:update`, whether on my machine or in a Debian Jessie virtual machine (on which I can install all the requirements), gives this:
```
Dropping database schema...
Database schema dropped successfully!
ATTENTION: This operation should not be executed in a production environment.

Creating database schema...
Database schema created successfully!
  > loading [1] Acts\CamdramBackendBundle\DataFixtures\ORM\SocietyFixtures
PHP Fatal error:  Call to undefined function Foolz\SphinxQL\mysqli_init() in /home/bateman/Documents/University/ADC/Camdram/camdram/vendor/foolz/sphinxql-query-builder/src/Connection.php on line 167

                                                                             
  [Symfony\Component\Debug\Exception\UndefinedFunctionException]             
  Attempted to call function "mysqli_init" from namespace "Foolz\SphinxQL".  
                                                                             

camdram:database:update [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command>
```

-----
So, I tried to give it what it appears to want (MySQLi) by installing `php5-mysql`. Repeating `camdram:database:update` gives
```
                                                                                                     
  [Symfony\Component\Debug\Exception\ContextErrorException]                                          
  Warning: mysqli::real_connect(): (HY000/2003): Can't connect to MySQL server on '127.0.0.1' (111)  
                                                                                                     
```
I'm persistent, so I tried setting up the MySQL server as described on the wiki. Running `php app/console doctrine:migrations:migrate` gives:
![virtualbox_camdram host_15_06_2018_14_35_48](https://user-images.githubusercontent.com/5357642/41470787-96b53f8e-70a9-11e8-9b04-b89adf776337.png)

I'm now stuck.

**Comment by hoyes** [comment]
Sorry about this George. Looks like ease of setting up a fresh checkout has regressed over the years in the name of fire-fighting bugs...

The short answer is we rely on Sphinx in production, but in order to ease setting up a dev checkout I originally tried to make it so it falls back to doing search directly on the DB if Sphinx isn't present, so the README doesn't mention Sphinx. But as fewer and fewer people actually created dev checkouts over the years this codepath has regressed. I've put more comments about wider Sphinx integration issues on #398.

I'll have a look and see if there's an easy way to fix this over the weekend. In the meantime, you may or may not have mileage trying out https://github.com/camdram/camdram/wiki/Sphinx-setup-guide to set up Sphinx.

**Comment by CHTJonas** [comment]
Should we not be targeting PHP7 by now? I notice that Travis seems to be building for both 5 & 7 so maybe just a case of comprehensively updating the README and the wiki. Not sure if building for Debian 10 will result in general oddness as it's only the testing release atm...

**Comment by hoyes** [comment]
We really really should but we've been geologically slow in getting there. PHP 7 is supported (and is on Travis to ensure we don't break compatibility in the meantime), but we're still running PHP 5 (5.6, but until fairly recently 5.3) on Tiberius, for reasons which I'll pick up on webteam.

**Comment by hoyes** [comment]
I *think* I've now done enough of #398 to make the Sphinx errors go away. I've tried to make everything gracefully degrade if Elasticsearch is not present, i.e.:
 * Search UI returns an error
 * There's a config parameter `search_enable_listeners` which is off on new checkouts, meaning you can edit shows etc without getting errors if it can't sync the changes to the search index. 

I've also just done a quick pass of updates to the README and the Wiki pages, and added a Travis test that runs most of the steps in the README and fails if they regress. The hope is that the steps should now be follow-able again in a recent version of Debian/Ubuntu.

@CHTJonas and @GKFX - it would be great if you could have another go at this in the near future and let us know what issues still remain.

**Comment by CHTJonas** [comment]
I get the following warnings now on a fresh checkout/install:
```
Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
Package symfony/assetic-bundle is abandoned, you should avoid using it. Use symfony/webpack-encore-pack instead.
```

**Comment by hoyes** [comment]
That's normal (for now...):

The symfony/assetic-bundle one is basically #397 

I haven't looked much into guzzle/guzzle but I suspect it's something depended on by another package. We're currently on old-ish versions of lots of packages because of PHP 5.6 so I'm hoping that once we're on PHP 7 we can upgrade stuff and that'll go away.

**Comment by CHTJonas** [comment]
I can also happily confirm that apart from those two warnings, Camdram checks-out and runs perfectly OK on an Apple Mac 😄! I seem to be using the operating system bundled version of PHP however I might have a few extra SQLite libs tagged on from [homebrew](https://brew.sh).

**Comment by hoyes** [comment]
Woop!

You might have noticed I just deleted some Mac stuff from the README becuase I had no way of verifying it and didn't realise you were a Mac user. Feel free to add anything useful.

Can hopefully tackle the two warnings in the coming weeks.

**Comment by hoyes** [comment]
I have just made some tweaks so that a working Camdram checkout can now be made in a single line, DB and all:

    composer create-project camdram/camdram camdram dev-master --no-interaction

It prints a (hopefully) friendly welcome message at the end of it all (see the commits above) to hopefully give a bit more guidance about what to do next. Thoughts and wording tweaks welcome.

And I've just refactored the README to use this method, as it's what Symfony itself now recommends. The idea is that you now get a working site by step 3, so there's a bit more instant gratification over telling people to create a fork first etc. I've also
 * Tried to fill in some of the gaps in the fork/commit/pull request workflow.
 * Removed some extraneous stuff that I don't think is applicable in PHP 7 (timezones and memory limits)
 * Removed the advice to install Composer by downloading and running a random script as Composer no longer seem to list this approach. Composer seems to be widely available in package managers these days (and is in Debian stretch) so have said to use that instead, plus a link to other installation methods on Composer's website.

**Comment by hoyes** [comment]
The README and dev checkout process should hopefully now be much more streamlined than when this ticket was created. Interested in a fresh perspective from @GKFX though before closing in case there's anything that I've missed through (being all too familiar with how everything's meant to work...)

**Comment by GKFX** [comment]
Sorry for the delays (and for being incapable of installing anything!). The instructions in the README nearly worked (Debian Stretch VM); I just had to install `php-mbstring`, `php-xml` and (optional, but got rid of warnings and made it faster) `php-zip`. Let me know what OS you used when re-writing the README and I'll make an appropriate note. I got the Camdram server running, and everything was fine.

However, when I moved on to https://github.com/camdram/camdram/wiki/Setting-up-a-MySQL-database, it didn't really work. Output of `php app/console doctrine:migrations:migrate` is now
```
                                                              
                    Application Migrations                    
                                                              

Migrating up to 20180705143359 from 20141208012213

  ++ migrating 20141228044056

     -> ALTER TABLE acts_support CHANGE supportid supportid INT DEFAULT NULL
     -> UPDATE acts_support AS s LEFT JOIN acts_support AS u ON s.supportid = u.id SET s.supportid = NULL WHERE u.id IS NULL
     -> ALTER TABLE acts_support ADD CONSTRAINT FK_A6F619725B919408 FOREIGN KEY (supportid) REFERENCES acts_support (id)
Migration 20141228044056 failed during Execution. Error An exception occurred while executing 'ALTER TABLE acts_support ADD CONSTRAINT FK_A6F619725B919408 FOREIGN KEY (supportid) REFERENCES acts_support (id)':

SQLSTATE[HY000]: General error: 1005 Can't create table `camdram`.`#sql-1ed_4` (errno: 121 "Duplicate key on write or update")

                                                                                                                                                             
  [Doctrine\DBAL\Exception\DriverException]                                                                                                                  
  An exception occurred while executing 'ALTER TABLE acts_support ADD CONSTRAINT FK_A6F619725B919408 FOREIGN KEY (supportid) REFERENCES acts_support (id)':  
                                                                                                                                                             
  SQLSTATE[HY000]: General error: 1005 Can't create table `camdram`.`#sql-1ed_4` (errno: 121 "Duplicate key on write or update")                             
                                                                                                                                                             

                                                                                                                                  
  [Doctrine\DBAL\Driver\PDOException]                                                                                             
  SQLSTATE[HY000]: General error: 1005 Can't create table `camdram`.`#sql-1ed_4` (errno: 121 "Duplicate key on write or update")  
                                                                                                                                  

                                                                                                                                  
  [PDOException]                                                                                                                  
  SQLSTATE[HY000]: General error: 1005 Can't create table `camdram`.`#sql-1ed_4` (errno: 121 "Duplicate key on write or update")  
                                                                                                                                  

doctrine:migrations:migrate [--write-sql] [--dry-run] [--query-time] [--allow-no-migration] [--configuration [CONFIGURATION]] [--db-configuration [DB-CONFIGURATION]] [--db DB] [--em EM] [--shard SHARD] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command> [<version>]

```
Deleting the database and starting again gives lots of positive looking test and then a similar error; difference is only that `#sql-1ed_4` becomes `#sql-884_3`. It seems to be the `acts_support` table every time; do you think this might relate to the switch to Brimir?

**Comment by hoyes** [comment]
Thanks very much for that. The install command was based on Ubuntu 18.04.

Looks like the older DB migrations are broken... As of the release just now the production DB is fully up to date so I'll take the opportunity to squash them down to one migration which should fix that... standby...

**Comment by hoyes** [comment]
I have just merged the Debian/Ubuntu lines in the README... turned out I already had those additional packages installed on my copy of Ubuntu so it was just an oversight on my part.

I've also just flattened the DB migrations, and made the new single DB migration run on a blank DB on Travis so we're more likely to detect if that breaks again. So hopefully the MySQL instructions will work for you now (although the previous go probably left the DB in an indeterminate state so might need to delete and recreate it).
